sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,23 @@ def reduce_inequalities(inequalities, symbols=[]):
 
     # restore original symbols and return
     return rv.xreplace({v: k for k, v in recast.items()})
+
+
+def reduce_boolean_expr(boolean_expr,symbols=[]):
```

Comment:
I think this should not be looking for And or Or in the expression, it should use `to_cnf` to convert to a standardized form free of Not and then act on whether the result is an And or Or as hinted in my comment on the issue page.